### PR TITLE
Hardening/1851 csub cache simplification

### PR DESCRIPTION
**Pull request by kzangeli**
Part of issue #1851.
Changed the field NotifyConditionVector of CachedSubscription to be a vector of strings.

Two doubts still in the code ...

### CODE REVIEW

File: src/lib/cache/subCache.cpp [hunk 1]
```diff
@@ -690,57 +683,33 @@ void subCacheItemInsert(CachedSubscription* cSubP)
...
+  bool                             blacklist
 )
 {
   //
   // Add the subscription to the subscription cache.
-  // But only if any of the entities in entityIdVector is pattern based -
-  // AND it is a subscription of type ONCHANGE
-  //
-
+  // But only if any of the entities in entityIdVector is pattern based - is this true?
```
Comment (kzangeli) [inline src/lib/cache/subCache.cpp]: Doubt 1

File: src/lib/cache/subCache.cpp [hunk 2]
```diff
@@ -765,7 +734,16 @@ void subCacheItemInsert
   cSubP->blacklist             = blacklist;
   cSubP->httpInfo              = httpInfo;
 
+  // cSubP->notifyConditionV = notifyConditionV; ???
```
Comment (kzangeli) [inline src/lib/cache/subCache.cpp]: Doubt 2

In C++ you can copy vectors like the comment proposes, right?

File: src/lib/cache/subCache.cpp (hunk 1 above)
Comment (fgalan) [inline src/lib/cache/subCache.cpp]: Not fully sure... I think that at the begining, csub cache was only used for pattern subscriptions. But at the end it was used for all subscriptions.

File: src/lib/cache/subCache.cpp (hunk 2 above)
Comment (fgalan) [inline src/lib/cache/subCache.cpp]: Maybe this helps: http://www.cplusplus.com/reference/vector/vector/operator=/

File: src/lib/mongoBackend/mongoSubCache.cpp [hunk 3]
```diff
@@ -500,7 +482,6 @@ void mongoSubCacheRefresh(const std::string& database)
     }
 
     int r = mongoSubCacheItemInsert(tenant.c_str(), sub);
-
     if (r == 0)
     {
       ++subNo;
```
Comment (fgalan) [inline src/lib/mongoBackend/mongoSubCache.cpp]: Is the change to .DISABLED of the custom notification test intended in this PR?

File: src/lib/cache/subCache.cpp (hunk 1 above)
Comment (fgalan) [inline src/lib/cache/subCache.cpp]: In fact, I guess that the subscrpition refresh logic is taking all the subscriptions or many .test will fail...

Comment (fgalan) [inline src/lib/cache/subCache.cpp]: I found it. From 0.26.0 changelog (https://github.com/telefonicaid/fiware-orion/releases/tag/0.26.0):

> Add: Unpatterned subscriptions now also in subscription cache (Issue #1475)

Comment (kzangeli) [inline src/lib/cache/subCache.cpp]: Good. That's what I thought.
Removing that line

File: src/lib/mongoBackend/mongoSubCache.cpp (hunk 3 above)
Comment (kzangeli) [inline src/lib/mongoBackend/mongoSubCache.cpp]: Yeah, I added that also after yet another scare of a failing functest.
We decided this yesterday so I included it here in this PR.

File: src/lib/cache/subCache.cpp (hunk 2 above)
Comment (kzangeli) [inline src/lib/cache/subCache.cpp]: It helps.
Changing the loop for the one-liner.

File: src/lib/mongoBackend/mongoSubCache.cpp (hunk 3 above)
Comment (fgalan) [inline src/lib/mongoBackend/mongoSubCache.cpp]: I prefer to defer the decision about this and do it in an independent PR (probably the very last one just before releasing 1.3.0 :))

File: src/lib/cache/subCache.cpp (hunk 2 above)
Comment (kzangeli) [inline src/lib/cache/subCache.cpp]: Fixed in dc432d8

File: src/lib/cache/subCache.cpp (hunk 1 above)
Comment (kzangeli) [inline src/lib/cache/subCache.cpp]: Fixed in dc432d8

File: src/lib/mongoBackend/mongoSubCache.cpp (hunk 3 above)
Comment (kzangeli) [inline src/lib/mongoBackend/mongoSubCache.cpp]: Ehh .... ok ...

Comment (kzangeli) [inline src/lib/mongoBackend/mongoSubCache.cpp]: Enabled again in 46412bc

**Comment by fgalan** [comment]
LGTM
